============================= test session starts ==============================
platform linux -- Python 3.8.13, pytest-6.2.2, py-1.11.0, pluggy-0.13.1 -- /home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/bin/python
cachedir: .pytest_cache
rootdir: /home/tdk/codes/courses/mlops_udacity/course_2/udacity_pipeline_exercise_14/genre_classification/check_data
collecting ... collected 4 items

test_data.py::test_column_presence_and_type ERROR
test_data.py::test_class_names ERROR
test_data.py::test_column_ranges ERROR
test_data.py::test_kolmogorov_smirnov ERROR

==================================== ERRORS ====================================
_______________ ERROR at setup of test_column_presence_and_type ________________

args = (<wandb.apis.public.Api object at 0x7f5418c0d460>,)
kwargs = {'name': 'exercise_14/preprocessed_data.csv:latest', 'type': None}
message = 'Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'
payload = 'Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'

    @wraps(func)
    def wrapper(*args, **kwargs):
        message = "Whoa, you found a bug."
        try:
>           return func(*args, **kwargs)

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:24: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <wandb.apis.public.Api object at 0x7f5418c0d460>
name = 'exercise_14/preprocessed_data.csv:latest', type = None

    @normalize_exceptions
    def artifact(self, name, type=None):
        """
        Returns a single artifact by parsing path in the form `entity/project/run_id`.
    
        Arguments:
            name: (str) An artifact name. May be prefixed with entity/project. Valid names
                can be in the following forms:
                    name:version
                    name:alias
                    digest
            type: (str, optional) The type of artifact to fetch.
        Returns:
            A `Artifact` object.
        """
        if name is None:
            raise ValueError("You must specify name= to fetch an artifact.")
        entity, project, artifact_name = self._parse_artifact_path(name)
>       artifact = Artifact(self.client, entity, project, artifact_name)

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:520: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>
client = <wandb.apis.public.RetryingClient object at 0x7f54abc77af0>
entity = 'kaushaltak', project = 'exercise_14'
name = 'preprocessed_data.csv:latest', attrs = None

    def __init__(self, client, entity, project, name, attrs=None):
        self.client = client
        self._entity = entity
        self._project = project
        self._artifact_name = name
        self._attrs = attrs
        if self._attrs is None:
>           self._load()

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:2583: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>

    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
E           ValueError: Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:3097: ValueError

During handling of the above exception, another exception occurred:

request = <SubRequest 'data' for <Function test_column_presence_and_type>>

    @pytest.fixture(scope="session")
    def data(request):
    
        reference_artifact = request.config.option.reference_artifact
    
        if reference_artifact is None:
            pytest.fail("--reference_artifact missing on command line")
    
        sample_artifact = request.config.option.sample_artifact
    
        if sample_artifact is None:
            pytest.fail("--sample_artifact missing on command line")
    
>       local_path = run.use_artifact(reference_artifact).file()

conftest.py:28: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/sdk/wandb_run.py:1957: in use_artifact
    artifact = public_api.artifact(type=type, name=name)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:62: in wrapper
    six.reraise(CommError, CommError(message, err), sys.exc_info()[2])
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/six.py:718: in reraise
    raise value.with_traceback(tb)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:24: in wrapper
    return func(*args, **kwargs)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:520: in artifact
    artifact = Artifact(self.client, entity, project, artifact_name)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:2583: in __init__
    self._load()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>

    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
E           wandb.errors.error.CommError: Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:3097: CommError
______________________ ERROR at setup of test_class_names ______________________

args = (<wandb.apis.public.Api object at 0x7f5418c0d460>,)
kwargs = {'name': 'exercise_14/preprocessed_data.csv:latest', 'type': None}
message = 'Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'
payload = 'Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'

    @wraps(func)
    def wrapper(*args, **kwargs):
        message = "Whoa, you found a bug."
        try:
>           return func(*args, **kwargs)

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:24: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <wandb.apis.public.Api object at 0x7f5418c0d460>
name = 'exercise_14/preprocessed_data.csv:latest', type = None

    @normalize_exceptions
    def artifact(self, name, type=None):
        """
        Returns a single artifact by parsing path in the form `entity/project/run_id`.
    
        Arguments:
            name: (str) An artifact name. May be prefixed with entity/project. Valid names
                can be in the following forms:
                    name:version
                    name:alias
                    digest
            type: (str, optional) The type of artifact to fetch.
        Returns:
            A `Artifact` object.
        """
        if name is None:
            raise ValueError("You must specify name= to fetch an artifact.")
        entity, project, artifact_name = self._parse_artifact_path(name)
>       artifact = Artifact(self.client, entity, project, artifact_name)

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:520: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>
client = <wandb.apis.public.RetryingClient object at 0x7f54abc77af0>
entity = 'kaushaltak', project = 'exercise_14'
name = 'preprocessed_data.csv:latest', attrs = None

    def __init__(self, client, entity, project, name, attrs=None):
        self.client = client
        self._entity = entity
        self._project = project
        self._artifact_name = name
        self._attrs = attrs
        if self._attrs is None:
>           self._load()

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:2583: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>

    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
E           ValueError: Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:3097: ValueError

During handling of the above exception, another exception occurred:

request = <SubRequest 'data' for <Function test_column_presence_and_type>>

    @pytest.fixture(scope="session")
    def data(request):
    
        reference_artifact = request.config.option.reference_artifact
    
        if reference_artifact is None:
            pytest.fail("--reference_artifact missing on command line")
    
        sample_artifact = request.config.option.sample_artifact
    
        if sample_artifact is None:
            pytest.fail("--sample_artifact missing on command line")
    
>       local_path = run.use_artifact(reference_artifact).file()

conftest.py:28: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/sdk/wandb_run.py:1957: in use_artifact
    artifact = public_api.artifact(type=type, name=name)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:62: in wrapper
    six.reraise(CommError, CommError(message, err), sys.exc_info()[2])
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/six.py:718: in reraise
    raise value.with_traceback(tb)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:24: in wrapper
    return func(*args, **kwargs)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:520: in artifact
    artifact = Artifact(self.client, entity, project, artifact_name)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:2583: in __init__
    self._load()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>

    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
E           wandb.errors.error.CommError: Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:3097: CommError
_____________________ ERROR at setup of test_column_ranges _____________________

args = (<wandb.apis.public.Api object at 0x7f5418c0d460>,)
kwargs = {'name': 'exercise_14/preprocessed_data.csv:latest', 'type': None}
message = 'Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'
payload = 'Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'

    @wraps(func)
    def wrapper(*args, **kwargs):
        message = "Whoa, you found a bug."
        try:
>           return func(*args, **kwargs)

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:24: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <wandb.apis.public.Api object at 0x7f5418c0d460>
name = 'exercise_14/preprocessed_data.csv:latest', type = None

    @normalize_exceptions
    def artifact(self, name, type=None):
        """
        Returns a single artifact by parsing path in the form `entity/project/run_id`.
    
        Arguments:
            name: (str) An artifact name. May be prefixed with entity/project. Valid names
                can be in the following forms:
                    name:version
                    name:alias
                    digest
            type: (str, optional) The type of artifact to fetch.
        Returns:
            A `Artifact` object.
        """
        if name is None:
            raise ValueError("You must specify name= to fetch an artifact.")
        entity, project, artifact_name = self._parse_artifact_path(name)
>       artifact = Artifact(self.client, entity, project, artifact_name)

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:520: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>
client = <wandb.apis.public.RetryingClient object at 0x7f54abc77af0>
entity = 'kaushaltak', project = 'exercise_14'
name = 'preprocessed_data.csv:latest', attrs = None

    def __init__(self, client, entity, project, name, attrs=None):
        self.client = client
        self._entity = entity
        self._project = project
        self._artifact_name = name
        self._attrs = attrs
        if self._attrs is None:
>           self._load()

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:2583: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>

    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
E           ValueError: Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:3097: ValueError

During handling of the above exception, another exception occurred:

request = <SubRequest 'data' for <Function test_column_presence_and_type>>

    @pytest.fixture(scope="session")
    def data(request):
    
        reference_artifact = request.config.option.reference_artifact
    
        if reference_artifact is None:
            pytest.fail("--reference_artifact missing on command line")
    
        sample_artifact = request.config.option.sample_artifact
    
        if sample_artifact is None:
            pytest.fail("--sample_artifact missing on command line")
    
>       local_path = run.use_artifact(reference_artifact).file()

conftest.py:28: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/sdk/wandb_run.py:1957: in use_artifact
    artifact = public_api.artifact(type=type, name=name)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:62: in wrapper
    six.reraise(CommError, CommError(message, err), sys.exc_info()[2])
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/six.py:718: in reraise
    raise value.with_traceback(tb)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:24: in wrapper
    return func(*args, **kwargs)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:520: in artifact
    artifact = Artifact(self.client, entity, project, artifact_name)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:2583: in __init__
    self._load()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>

    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
E           wandb.errors.error.CommError: Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:3097: CommError
__________________ ERROR at setup of test_kolmogorov_smirnov ___________________

args = (<wandb.apis.public.Api object at 0x7f5418c0d460>,)
kwargs = {'name': 'exercise_14/preprocessed_data.csv:latest', 'type': None}
message = 'Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'
payload = 'Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"'

    @wraps(func)
    def wrapper(*args, **kwargs):
        message = "Whoa, you found a bug."
        try:
>           return func(*args, **kwargs)

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:24: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <wandb.apis.public.Api object at 0x7f5418c0d460>
name = 'exercise_14/preprocessed_data.csv:latest', type = None

    @normalize_exceptions
    def artifact(self, name, type=None):
        """
        Returns a single artifact by parsing path in the form `entity/project/run_id`.
    
        Arguments:
            name: (str) An artifact name. May be prefixed with entity/project. Valid names
                can be in the following forms:
                    name:version
                    name:alias
                    digest
            type: (str, optional) The type of artifact to fetch.
        Returns:
            A `Artifact` object.
        """
        if name is None:
            raise ValueError("You must specify name= to fetch an artifact.")
        entity, project, artifact_name = self._parse_artifact_path(name)
>       artifact = Artifact(self.client, entity, project, artifact_name)

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:520: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>
client = <wandb.apis.public.RetryingClient object at 0x7f54abc77af0>
entity = 'kaushaltak', project = 'exercise_14'
name = 'preprocessed_data.csv:latest', attrs = None

    def __init__(self, client, entity, project, name, attrs=None):
        self.client = client
        self._entity = entity
        self._project = project
        self._artifact_name = name
        self._attrs = attrs
        if self._attrs is None:
>           self._load()

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:2583: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>

    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
E           ValueError: Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:3097: ValueError

During handling of the above exception, another exception occurred:

request = <SubRequest 'data' for <Function test_column_presence_and_type>>

    @pytest.fixture(scope="session")
    def data(request):
    
        reference_artifact = request.config.option.reference_artifact
    
        if reference_artifact is None:
            pytest.fail("--reference_artifact missing on command line")
    
        sample_artifact = request.config.option.sample_artifact
    
        if sample_artifact is None:
            pytest.fail("--sample_artifact missing on command line")
    
>       local_path = run.use_artifact(reference_artifact).file()

conftest.py:28: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/sdk/wandb_run.py:1957: in use_artifact
    artifact = public_api.artifact(type=type, name=name)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:62: in wrapper
    six.reraise(CommError, CommError(message, err), sys.exc_info()[2])
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/six.py:718: in reraise
    raise value.with_traceback(tb)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/normalize.py:24: in wrapper
    return func(*args, **kwargs)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:520: in artifact
    artifact = Artifact(self.client, entity, project, artifact_name)
/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:2583: in __init__
    self._load()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not subscriptable") raised in repr()] Artifact object at 0x7f5418c0d7f0>

    def _load(self):
        query = gql(
            """
        query Artifact(
            $entityName: String!,
            $projectName: String!,
            $name: String!
        ) {
            project(name: $projectName, entityName: $entityName) {
                artifact(name: $name) {
                    ...ArtifactFragment
                }
            }
        }
        %s
        """
            % ARTIFACT_FRAGMENT
        )
        response = None
        try:
            response = self.client.execute(
                query,
                variable_values={
                    "entityName": self.entity,
                    "projectName": self.project,
                    "name": self._artifact_name,
                },
            )
        except Exception:
            # we check for this after doing the call, since the backend supports raw digest lookups
            # which don't include ":" and are 32 characters long
            if ":" not in self._artifact_name and len(self._artifact_name) != 32:
                raise ValueError(
                    'Attempted to fetch artifact without alias (e.g. "<artifact_name>:v3" or "<artifact_name>:latest")'
                )
        if (
            response is None
            or response.get("project") is None
            or response["project"].get("artifact") is None
        ):
>           raise ValueError(
                'Project %s/%s does not contain artifact: "%s"'
                % (self.entity, self.project, self._artifact_name)
            )
E           wandb.errors.error.CommError: Project kaushaltak/exercise_14 does not contain artifact: "preprocessed_data.csv:latest"

/home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/apis/public.py:3097: CommError
=============================== warnings summary ===============================
../../../../../../../anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/vendor/watchdog/utils/bricks.py:175
  /home/tdk/anaconda3/envs/mlflow-c86639da1eecf5a556ffdc9bec299b5662e80681/lib/python3.8/site-packages/wandb/vendor/watchdog/utils/bricks.py:175: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10